axt/callSnps.go

```diff
@@ -21,33 +21,33 @@ func AxtVcfToFile(filename string, axtList []*Axt, fa []*fasta.Fasta) {
 func AxtToVcf(axtFile *Axt) []*vcf.Vcf {
 	var answer []*vcf.Vcf
 	var curr *vcf.Vcf
-	rCount := axtFile.RStart - 1
+	var rCount int = int(axtFile.RStart) - 1
 	qCount := axtFile.QStart - 1
 	for i := 0; i < len(axtFile.RSeq); i++ {
 		if axtFile.RSeq[i] != dna.Gap && axtFile.QSeq[i] != dna.Gap {
 			rCount++
 			qCount++
 			//snp mismatch
 			if dna.ToUpper(axtFile.RSeq[i]) != dna.ToUpper(axtFile.QSeq[i]) {
-				curr = &vcf.Vcf{Chr: axtFile.RName, Pos: rCount, Id: axtFile.QName, Ref: dna.BaseToString(dna.ToUpper(axtFile.RSeq[i])), Alt: dna.BaseToString(dna.ToUpper(axtFile.QSeq[i])), Qual: 30, Filter: "PASS", Info: fmt.Sprintf("query=%d", qCount), Format: "SVTYPE=SNP", Notes: AxtInfo(axtFile)}
+				curr = &vcf.Vcf{Chr: axtFile.RName, Pos: rCount, Id: axtFile.QName, Ref: dna.BaseToString(dna.ToUpper(axtFile.RSeq[i])), Alt: []string{dna.BaseToString(dna.ToUpper(axtFile.QSeq[i]))}, Qual: 30, Filter: "PASS", Info: fmt.Sprintf("query=%d;SVTYPE=SNP;%s", qCount, AxtInfo(axtFile))}
 				answer = append(answer, curr)
 			}
 		}
 		//insertion in VCF record
 		if axtFile.RSeq[i] == dna.Gap {
 
 			qCount++
-			curr = &vcf.Vcf{Chr: axtFile.RName, Pos: rCount, Id: axtFile.QName, Ref: dna.BaseToString(dna.ToUpper(axtFile.RSeq[i-1])), Alt: dna.BaseToString(dna.ToUpper(axtFile.QSeq[i-1])), Qual: 24, Filter: "PASS", Info: fmt.Sprintf("query=%d", qCount), Format: "SVTYPE=INS", Notes: AxtInfo(axtFile)}
+			curr = &vcf.Vcf{Chr: axtFile.RName, Pos: rCount, Id: axtFile.QName, Ref: dna.BaseToString(dna.ToUpper(axtFile.RSeq[i-1])), Alt: []string{dna.BaseToString(dna.ToUpper(axtFile.QSeq[i-1]))}, Qual: 24, Filter: "PASS", Info: fmt.Sprintf("query=%d;SVTYPE=SNP;%s", qCount, AxtInfo(axtFile))}
 
 			for j := i; j < len(axtFile.RSeq); j++ {
 				if dna.ToUpper(axtFile.RSeq[j]) == dna.Gap {
-					curr.Alt += dna.BaseToString(dna.ToUpper(axtFile.QSeq[j]))
+					curr.Alt = append(curr.Alt, dna.BaseToString(dna.ToUpper(axtFile.QSeq[j])))
```

Comment:
not sure if we should append this one by one. What is happening is we are looping forward and checking if there are bases for the insertion/deletion. It should follow the same `+=` logic.